apps/chat/src/components/Chat/ModelsDialog.tsx

```diff
@@ -139,18 +166,36 @@ export const ModelsDialog: FC<ModelsDialogProps> = ({
         </button>
       </div>
 
-      <div className="px-3 md:px-5">
+      <div className="relative flex px-3 md:px-5">
         <input
           name="titleInput"
-          placeholder={t('Search model, assistant or application') || ''}
+          placeholder={t('Search model, assistant, or application') || ''}
           type="text"
           onChange={(e) => {
             handleSearch(e.target.value);
           }}
-          className="m-0 w-full rounded border border-primary bg-transparent px-3 py-2 outline-none placeholder:text-secondary focus-visible:border-accent-primary"
+          className="m-0 w-full grow rounded border border-primary bg-transparent px-3 py-2 outline-none placeholder:text-secondary focus-visible:border-accent-primary"
         ></input>
+        {isCustomApplicationsEnabled ? (
+          <button
+            onClick={openAddApplicationModal}
+            className="absolute right-3 flex h-full w-[100px] items-center gap-2 text-accent-primary"
+          >
+            <IconLayoutGridAdd height={18} width={18} />
```

Comment:
nit: you can use `size` prop here instead of `height` and `width`
and be careful about shrinking